trillian/docs/Operation.md

```diff
@@ -15,36 +15,158 @@ result in the [failure](https://tools.ietf.org/html/rfc6962#section-7.3) of a CT
+
+This repository includes a couple of tools to help with this testing.  Firstly,
+the
+[`preloader` tool](https://github.com/google/certificate-transparency-go/blob/master/preload/preloader)
+allows the contents of a source log to be copied into a destination log.  This
+tool has command-line options to control its parallelism, but is fundamentally
+a single-process executable.
+
+The other load-testing tool is the
+[`ct_hammer`](https://github.com/google/certificate-transparency-go/blob/master/trillian/integration/ct_hammer),
+which tests all of the
+[RFC 6962 entrypoints](https://tools.ietf.org/html/rfc6962#section-4) with both
+valid and invalid inputs.
+
+ - For write-path testing, `ct_hammer` relies on the Log under test being
+   configured to accept a test root certificate, so that synthetic test
+   certificates can be submitted.
+ - For convenience, `ct_hammer` accepts the same format of configuration file
+   that is used to configure the CTFE.  (However, be careful not to distribute
+   a CTFE configuration file that includes non-test
+   [private keys](#key-management).)
+ - The `--rate_limit` option controls the overall rate limit for the tool.
+ - Multiple instances of `ct_hammer` can be run in parallel to allow load
+   testing to be scaled up arbitrarily.
+
+These testing tools can also be used to confirm that the Log continues to
+operate normally while various maintenance activities &ndash; software
+rollouts, machine turndowns, configuration updates, etc. &ndash; are in
+progress.
+
+
+## Backups
+
+For most production systems with persistent data, regular backups are
+recommended.  However, the cryptographic nature of a CT Log means that backups
+of its data induce a dangerous temptation.
+
+The temptation is this: if you have a backup, at some point you will feel the
+urge to perform a **restore** from backup.  If any data has been accepted for
+inclusion since that backup (and a signed promise-to-include issued), then
+restoring the backup is effectively forking the underlying Merkle tree.  This
+breaks the tree's append-only property &ndash; resulting in log
+disqualification.
 
 
 ## Troubleshooting
 
-TODO(daviddrysdale): expand this
- - Any MySQL tools to explore the database? Safe to do so for a running log?
- - Changing logging levels? What to look for in logs, and where?
- - Backups?  turn out not to be not helpful -- restoring a database is likely
-   to fork the Merkle tree.
+All of the Trillian and CTFE binaries use the
+[glog](https://github.com/golang/glog) library for logging, so additional
+diagnostic information can be obtained by modifying the glog options, for
+example, by enabling `--logtostderr -v 1`.
+
+(Another useful glog option for debugging particular problems is the
+`--log_backtrace_at` option, which triggers glog to emit a full stack trace at
+particular logging statements.)
+
+Also, the underlying storage system can be queried independently, using the
```

Comment:
Also, where possible, ..., using the appropriate vendor tool (e.g. MySQL command line client, or Cloud Spanner web interface, etc.)